### ISSUE TITLE
Converting networkX graph to DGL graph object

### ISSUE THREAD

**Issue by shakeel608** [dmlc/dgl#6178]
## 📚 Documentation

When I convert my networkX graph to DGL graph, why I get double the number of edges in DGL object.

How to fix it ? 

Any help would be appreciated please ?

**Comment by Rhett-Ying** [comment]
could you share a minimum reproducible code?

**Comment by shakeel608** [comment]
`def load_graph(self, path):
        #Load graph
        with open(path, 'r') as f:
            graph_load = json.loads(f.read())
            return nx.readwrite.json_graph.node_link_graph(graph_load)`

graph_nx = load_graph(graphs_path)
dgl_graph = dgl.from_networkx(graph_nx, node_attrs=['features', 'label'])

**Comment by Rhett-Ying** [comment]
`DGLGraph` is always directed. You could use `dgl.add_reverse_edges()` or `dgl.to_bidirected()` to see if any new edges are added. If not, then the DGLGraph is already bi-directed. Or try with `directed=False` in `node_link_graph` of nx.

**Comment by shakeel608** [comment]
Thank you so much 
But I have another questions
I tried a sample code 
Suppose 
I have a graph with adj matrix as given below 

`graph.adjacency_matrix().to_dense()`

`tensor([[0., 1.],
              [1., 0.]])`

When I just print the properties of variable 'graph', it gives me following 

`Graph(num_nodes=2, num_edges=4,
      ndata_schemes={'h': Scheme(shape=(), dtype=torch.float32)}
      edata_schemes={})`

Why num_edges is 4. Shouldn't it be only two even if the DGL graphs are directed

**Comment by Rhett-Ying** [comment]
that's weird. could you share the minimum reproducible code for above case? Below is the one for illustration.
```
import dgl
import torch

g = dgl.graph(([0, 1], [1, 0]))
print(g.adjacency_matrix().to_dense())
print(g)
```

```
tensor([[0., 1.],
        [1., 0.]])
Graph(num_nodes=2, num_edges=2,
      ndata_schemes={}
      edata_schemes={})
```
